stdlib/LinearAlgebra/test/dense.jl

```diff
@@ -395,6 +395,19 @@ end
     @test kron(b',2) == [8 10 12]
 end
 
+@testset "kron!" begin
+    a = [1.0, 0.0]
+    b = [0.0, 1.0]
+    @test kron!([1.0, 0.0], b, 0.5) == [0.0; 0.5]
+    b = [0.0, 1.0]
+    c = Vector{Float64}(undef, 4)
+    c = kron!(c, a, b)
```

Comment:
Since `kron!` is in-place, this call looks a bit strange, but we could make it a return value test:
```
    d = kron!(c, a, b)
    @test d === c
```
or simply replace it with
```suggestion
    kron!(c, a, b)
```
? I guess we don't test the return value usually, so the shorter call may be more appropriate.